query/src/main/java/org/infinispan/query/Search.java

```diff
@@ -71,7 +73,7 @@ public static QueryFactory getQueryFactory(Cache<?, ?> cache) {
       }
       checkBulkReadPermission(advancedCache);
       QueryEngine<?> queryEngine = ComponentRegistryUtils.getEmbeddedQueryEngine(advancedCache);
-      return new EmbeddedQueryFactory(queryEngine);
+      return new EmbeddedQueryFactory(queryEngine, advancedCache.hasFlag(Flag.CACHE_MODE_LOCAL));
```

Comment:
this seems lot of changes to enable local mode. Wouldn't be easier just add an overload with a boolean to
`getQueryFactory(Cache<?, ?> cache, boolean localMode)` and
`getIndexer(Cache<K, V> cache, boolean localMode)`?

it looks like the cache instance is discarded anyway.

Reply:
idk @pruivo
It'd prevent the changes in core only AFAICT (which are small) and with the side effect of adding more methods in the API for a feature that is somehow advanced, more for troubleshooting and disaster recovery, the capacity of doing targeted queries and reindexes

Reply:
Plus I see honouring the already existent flag a more natural way to enable local when the query API moves to core.

Reply:
probably it is only me, but it does not feel very natural to me to do `getQueryFactory(cache.withFlag(CACHE_MODE_LOCAL))` if we want to query the local node. Especially there are no (java)docs that mention it.

Reply:
it will probably become ```cache.withFlag(CACHE_MODE_LOCAL).query("FROM Something)``` with the new API

Reply:
> it will probably become `cache.withFlag(CACHE_MODE_LOCAL).query("FROM Something)` with the new API

that looks better :)